Sources/AsyncQueue/ActorQueue.swift

```diff
@@ -0,0 +1,144 @@
+///         queue.async {
+///             await self.append(message)
+///         }
+///     }
+///
+///     nonisolated
+///     public func retrieveLogs() async -> [String] {
+///         await queue.await { await self.logs }
+///     }
+///
+///     private func append(_ message: String) {
+///         logs.append(message)
+///     }
+///
+///     private let queue = ActorQueue()
+///     private var logs = [String]()
+/// }
+/// ```
+///
+/// - Warning: Execution order is not guaranteed unless the enqueued tasks interact with a single `actor` instance.
+public final class ActorQueue {
+
+    // MARK: Initialization
+
+    /// Instantiates an actor queue.
+    /// - Parameter priority: The baseline priority of the tasks added to the asynchronous queue.
+    public init(priority: TaskPriority? = nil) {
+        var capturedTaskStreamContinuation: AsyncStream<@Sendable () async -> Void>.Continuation? = nil
+        let taskStream = AsyncStream<@Sendable () async -> Void> { continuation in
+            capturedTaskStreamContinuation = continuation
+        }
+        // Continuation will be captured during stream creation, so it is safe to force unwrap here.
+        // If this force-unwrap fails, something is fundamentally broken in the Swift runtime.
+        taskStreamContinuation = capturedTaskStreamContinuation!
+
+        Task.detached(priority: priority) {
+            let executor = ActorExecutor()
+            for await task in taskStream {
+                await executor.suspendUntilStarted(task)
+            }
+        }
+    }
+
+    deinit {
+        taskStreamContinuation.finish()
+    }
+
+    // MARK: Public
+
+    /// Schedules an asynchronous task for execution and immediately returns.
+    /// The scheduled task will not execute until all prior tasks have completed or suspended.
+    /// - Parameter task: The task to enqueue.
+    public func async(_ task: @escaping @Sendable () async -> Void) {
+        taskStreamContinuation.yield(task)
+    }
+
+    /// Schedules an asynchronous task and returns after the task is complete.
+    /// The scheduled task will not execute until all prior tasks have completed or suspended.
+    /// - Parameter task: The task to enqueue.
+    /// - Returns: The value returned from the enqueued task.
+    public func await<T>(_ task: @escaping @Sendable () async -> T) async -> T {
+        await withUnsafeContinuation { continuation in
+            taskStreamContinuation.yield {
+                continuation.resume(returning: await task())
+            }
+        }
+    }
+
+    /// Schedules an asynchronous throwing task and returns after the task is complete.
+    /// The scheduled task will not execute until all prior tasks have completed or suspended.
+    /// - Parameter task: The task to enqueue.
+    /// - Returns: The value returned from the enqueued task.
+    public func await<T>(_ task: @escaping @Sendable () async throws -> T) async throws -> T {
```

Comment:
If this function is accepts a throwing task should the above one accept a throwing task too?

Reply:
Ideally we'd use `rethrows` here so we only need to declare one method, however I couldn't get `rethrows` to work here: there's no way today (that I know of) to signal to the compiler that we're only going to `throw` within `withUnsafeThrowingContinuation` if and only if the `task` `throws`. So I created one method that can `throw` and one that can't.